ush/python/pygfs/task/stage_ic.py

```diff
@@ -1,58 +1,480 @@
 #!/usr/bin/env python3
+"""
+Stage Initial Conditions (IC) Task
 
+Overview
+--------
+This module constructs cycle and member-specific COM directory path variables
+required for initial conditions for the Stage IC task.
+"""
 import os
 from logging import getLogger
-from typing import Any, Dict
-
-from wxflow import FileHandler, Task, logit, parse_j2yaml
+from typing import Any, Dict, Tuple, Optional
+from datetime import timedelta
+from wxflow import FileHandler, Task, logit, parse_j2yaml, AttrDict, to_YMD, to_fv3time
 
 logger = getLogger(__name__.split('.')[-1])
 
 
 class Stage(Task):
-    """Task to stage initial conditions
-    """
+    """Task to stage initial conditions"""
 
     @logit(logger, name="Stage")
     def __init__(self, config: Dict[str, Any]) -> None:
         """Constructor for the Stage task
-        The constructor is responsible for collecting necessary settings based on
-        the runtime options and RUN.
 
         Parameters
         ----------
         config : Dict[str, Any]
             Incoming configuration for the task from the environment
+        """
+        super().__init__(config)
+
+    def _copy_base_config(self) -> Dict[str, Any]:
```

Comment:
```suggestion
    @logit(logger)
    def _copy_base_config(self) -> Dict[str, Any]:
```

Add `logit` decorator to all methods to allow tracing and debugging.